### PR TITLE
Changed query building order to play nice with forwarded calls

### PR DESCRIPTION
**Pull request by voydz**
PR for issue #100 

May conflict with #93 

This basically just changes the order of when internal search attributes and forwardedCalls being added to the query builder.

### CODE REVIEW

**Comment by freekmurze** [comment]
Could you rebase this against the current master?

**Comment by voydz** [comment]
@freekmurze no problem

do not mind the force push 👀 

I forgot to actually modify the added tests to reflect changes in #110

File: .php_cs.cache [hunk 1]
```diff
@@ -1 +1 @@
...
...g":true,"array_syntax":{"syntax":"short"},"ordered_imports":{"sortAlgorithm":"alpha"},"no_unused_imports":true,"not_operator_with_successor_space":true,"trailing_comma_in_multiline_array":true,"phpdoc_scalar":true,"unary_operator_spaces":true,"binary_operator_spaces":true,"blank_line_before_statement":{"statements":["break","continue","declare","return","throw","try"]},"phpdoc_single_line_var_spacing":true,"phpdoc_var_without_name":true},"hashes":{"src\/Searchable.php":1242980759,"src\/Search.php":1813740488,"src\/SearchResultCollection.php":2567122850,"src\/SearchAspect.php":4127695379,"src\/SearchResult.php":2693161934,"src\/SearchableAttribute.php":3752884723,"src\/Exceptions\/InvalidSearchableModel.php":3300456690,"src\/Exceptions\/InvalidModelSearchAspect.php":2689367197,"src\/ModelSearchAspect.php":4153366588,"tests\/stubs\/Account.php":470375312,"tests\/stubs\/CustomNameSearchAspect.php":2380198170,"tests\/SearchResultTest.php":4266329428,"tests\/Models\/TestComment.php":2445821388,"tests\/Models\/TestModel.php":2013473586,"tests\/TestCase.php":517075324,"tests\/SearchAspectTest.php":830066418,"tests\/ModelSearchAspectTest.php":2803194365,"tests\/SearchTest.php":2850208164}}
```
Comment (freekmurze) [inline .php_cs.cache:1]: This file may be deleted an ignored

**Review by freekmurze — CHANGES_REQUESTED** [review]

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**